# CSE110 Assignment 2 

This assignment is to guide in solving basic programming problems using loop in Python. 




##Write your name, student ID and theory section number below:


In [ ]:
#STUDENT NAME:ANIKA ISLAM 
#STUDENT ID:21101298
#CSE110 SECTION: 11

## Write the code in Python for following problems:

**<font color='red'>[MUST MAINTAIN VARIABLE NAMING CONVENTIONS FOR ALL THE TASKS]</font>**

### Task 1

Write python program, which prints the following sequences of values in loops:

a)	24, 18, 12, 6, 0, -6\
b)	-10, -5, 0, 5, 10, 15, 20\
c)	18, 27, 36, 45, 54, 63\
d)	18,-27, 36,-45,54,-63

=====================================================================

**Hints(1):** Use a while loop for solving these problems.

**Hints(2):** We are already familiar with the print() function. But when we use it to print any value it automatically adds an additional newline after each print statement.

For example:\
print(1)\
print(2)

Output:\
1\
2

=====================================================================

To solve this problem, in Python3, we need to add an extra argument (end = " ") in the print function which tells the program to skip printing the additional newline.

For example:\
print(1, end =" ")\
print(2)

Output:(prints the next output right to the previous one)\
12

=====================================================================

In Task-1(a), the loop counter should be initialized at 24 and the loop should terminate when the loop counter reaches -6. The difference between the first two values is 24-18=6. So the loop counter value is getting decremented by 6 in every iteration.

For your understanding task 1(a) code is done for you.


In [ ]:
# a) 24, 18, 12, 6, 0, -6

# initialize loop counter
counter = 24

# loop structure
while counter >= -6:
    
    #inside loop body
    if counter == -6:
        print(counter, end = "") 
    else:
        print(counter, end = ", ") 
     
    counter = counter - 6 #updating loop counter
    #inside loop body
    
#outside loop body

24, 18, 12, 6, 0, -6

In [1]:
(a) count = 24 
while count >= -6:
    if count == -6 :
        print(count, end = '')
    else:
        print(count, end = ',')
    count+=-6

24,18,12,6,0,-6

In [3]:
# b)-10, -5, 0, 5, 10, 15, 20
#initialize loop
counter=-10
#loop structure
while (counter<=20):
    if (counter==20):
        print(counter,end=' ')
    else:
        print(counter,end=' , ')
    counter=counter+5 #updating loop centre 
    #inside loop body
#outside loop body 

-10 , -5 , 0 , 5 , 10 , 15 , 20 

In [5]:
# b 
count = -10 
while count <= 20:
    if count == 20:
        print(count,end = '')
    else:
        print(count,end = ', ')
    count+=5

-10, -5, 0, 5, 10, 15, 20

In [4]:
# c)18, 27, 36, 45, 54, 63
#initializing loop 
counter=18
#loop structure 
while (counter<=63):
    if (counter==63):
        print(counter,end=' ')
    else:
        print(counter,end=' , ')
    counter=counter+9
    #inside loop body 
#outside loop body

18 , 27 , 36 , 45 , 54 , 63 

In [7]:
# c 
count = 18 
while count <= 63:
    if count == 63 :
        print(count, end = '')
    else:
        print(count, end = ', ')
    count+=9

18, 27, 36, 45, 54, 63

**Hints for 1(d):** 

print(5 * (-1)) gives output -5\
print("-" + str(5)) gives output -5

In [3]:
# d)18,-27, 36,-45,54,-63
#initializing loop
counter=18
#loop structure
while (counter<=63):
    if(counter%2==0):
        print(counter,end=' , ')
    else:
        if(counter<63):
            print(counter*(-1),end=' , ')
        else:
            print(counter*(-1),end='  ')
    counter=counter+9 #updatibg loop centre 
    #inside loop body
#outside loop body


18 , -27 , 36 , -45 , 54 , -63  

In [2]:
#d 
count = 18 
while count <= 63  :
    if count%2 == 0 :
        print(count,end=', ')
    else:
        if count < 63:
            print(count*(-1), end = ', ')
        else:
            print(count*(-1), end = ' ')
    count+=9

18, -27, 36, -45, 54, -63 

====================================================================\
<font color='blue'>
All the tasks can be solved using both the "while loop" and the "for loop". But you need to choose the best looping construct which is the most suitable for solving a particular task.
</font>

====================================================================

### Task 2
Write a Python code for the following:

1)	Ask the user to enter the name of his favorite car.\
2)	Ask the user to enter a Number

Display the name of the user’s favorite car, the number of times specified in the second step.

==========================================================

**Example 01:** If the user enters “Toyota” and 20, your program should print the name “Toyota” twenty times.

**Input:**\
Toyota\
2

**Output:**\
Toyota\
Toyota

==========================================================

**Example02:** If the user enters “Veyron” and 5, your program should print the name “Veyron” five times.

**Input:**\
Veyron\
5

**Output:**\
Veyron\
Veyron\
Veyron\
Veyron\
Veyron


In [8]:
#Todo
#taking name of car from user
name=input('enter name of car: ')
#taking number from user
number=int(input('enter number: '))
#initializing loop 
counter=1
#loop structure 
while(counter<=number):
    print(name)
    counter=counter+1 #updating loop centre 
    #inside loop body
#outside loop body 

enter name of car: toyota 
enter number: 2
toyota 
toyota 


In [9]:
car = input ('Enter name of favourite car: ')
num = int(input('Enter a number: '))
for i in range(0,num):
    print (car)

    

Enter name of favourite car: toyota
Enter a number: 2
toyota
toyota


### Task 3
Write a Python code of a program that adds all numbers that are multiples of **both 7 and 9** up to 600 (including 600).

Output: 2835


In [9]:
#Todo
#initializing loop 
counter=7
#taking sum as an empty integer 
sum=0
#loop structure
while(counter<=600):
    if(counter%7==0 and counter%9==0):
        sum=sum+counter
    counter=counter+1 #updating loop centre 
    #inside loop body 
print(sum)
#outside loop body         
    
    

2835


In [1]:
summ=0
for i in range(600):
    if (i%7==0) and (i%9==0):
        summ+=i
print(summ)

2835


### Task 4
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9** up to 600(including 600). \
Ensure that numbers like 63 are added only once in the sum.

Output: 42649


In [11]:
#Todo
#initializing the loop
counter=7
#taking sum as an empty integer 
sum=0
#loop structure 
while (counter<=600):
    if(counter%7==0 or counter%9==0):
        sum=sum+counter
    counter=counter+1 #updating loop centre 
    #inside loop body 
print(sum)
#outside loop body 
    
    

42649


In [2]:
summ=0
for i in range(600):
    if (i%7==0) or (i%9==0):
        summ+=i
print(summ)

42649


### Task 5
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9 but not both**, up to 600(including 600).

Output: 39814


In [12]:
#Todo
#initializing loop 
counter=1
#taking sum as an empty integer 
sum=0
#loop structure 
while(counter<=600):
    if(counter%7==0 and counter%9==0): #63%7==0 and 63%9==0
        sum=sum #sum is always 0
    elif(counter%7==0 or counter%9==0): #14%7==0 or 18%9==0
        sum=sum+counter #sum is only multiple of 7 or 9 
    counter=counter+1 #updating loop centre 
    #inside loop body 
print(sum)
#outside loop body 

39814


In [3]:
summ=0
for i in range(600):
    if (i%7==0) and (i%9==0):
        continue
    elif(i%7==0):
        summ+=i
    elif(i%9==0):
        summ+=i
print(summ)

39814


### Task 6
Write a Python code to displays all the **odd numbers** between 10 and 50.

Output: 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49


In [1]:
#Todo
#initializing loop 
counter=11
#loop structure
while (counter>=10 and counter<=50):
    if(counter%2!=0):
        print(counter,end=' ')
    else:
        print(counter)
    counter=counter+2 #updating loop centre
    #inside loop body 
#outside loop body




11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

In [5]:

for i in range(10,51):
    if (i%2!=0):
        print(i,end=' ')

11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

### Task 7
Write a Python code that will calculate the **value of y if the expression** of y is as follows (n is the input):  

$y = 1^2 - 2^2 +3^2 -4^2 +5^2 .........+ n^2 $


==========================================================

**Example01:** 

**Input:**\
10

**Output:**\
-55

==========================================================

**Example02:** 

**Input:**\
20

**Output:**\
-210



In [2]:
#Todo
#taking a number from user 
n=int(input('enter a number: '))
#taking y as an empty integer 
y=0
for i in range(0,n+1):
    if(i%2==0): #output is even and -ve 
        y=y-(i**2)
    else: #output is odd and +ve
        y=y+(i**2)
print(y)

enter a number: 20
-210


In [14]:
n = int(input('Enter a number: '))
y = 0 
for i in range (0,n+1):
    if i%2 == 0:
        y = y - (i**2)
    else:
        y = y + (i**2)
print(y)

Enter a number: 20
-210


### Task 8

Write a Python code of a program that asks the user to enter ten numbers then display the total and the average of **ONLY** the **odd numbers** among those ten numbers. <font color='red'>[Use of list is completely unnecesary for this task]</font>

==========================================================

**Example01:** 

**Input:**\
1

2

3

4

5

6

7

8

9

10

**Output:**
Total is 25 and Average is 5.0


**Explanation:**

Total is 25 = (1+3+5+7+9) and Average is 25/5 = 5.0

==========================================================


**Example02:** 

**Input:**\
-20

3

-5

40

-17

9

20

-8

99
  
-200

**Output:**

Total is 89 and Average is 17.8


**Explanation:**

Total is 89 =(3+(-5)+(-17)+9+99) and Average is 89/5 = 17.8



In [4]:
#Todo
#taking sum and average as empty integers
sum=0
average=0
counter=0
for i in range(0,10): #count starts from 0 and stops before 10 (0,1,2,3...9)
    number=int(input('enter a number: '))
    if(number%2!=0): #9%2!=0
        counter=counter+1
        sum=sum+number #sum
        average=sum/counter #average
print(sum)
print(average)

enter a number: -20
enter a number: 3
enter a number: -5
enter a number: 40
enter a number: -17
enter a number: 9
enter a number: 20
enter a number: -8
enter a number: 99
enter a number: -200
89
17.8


### Task 9
Write a Python code for the following:
* Ask the user to enter a Number, N
* Display the summation of multiples of 7 up to that number (**from 1 to N inclusive**)

==========================================================

**Example01:** 

**Input:**\
50

**Output:**\
196

**Explanation:**\
7 + 14 + 21 + 28 + 35 + 42 + 49 = 196

==========================================================

**Example02:** 

**Input:**\
75

**Output:**\
385

**Explanation:**\
7 + 14 + 21 + 28 + 35 + 42 + 49 + 56 + 63 + 70 = 385





In [28]:
#Todo
#taking a number from user
N=int(input('enter a number: '))
#taking sum as an empty integer
sum=0
#initializing loop 
counter=1
#loop structure
while (counter<=N):
    if(counter%7==0): #multiple of 7 
        sum=sum+counter #sum of multiple of 7
    counter=counter+1 #updating loop centre 
    #inside loop body 
print(sum)
#outside loop body

    


enter a number: 75
385


In [15]:
N = int(input('Enter a number: '))
summ=0
for i in range(0,N):
    if i%7 == 0: 
        summ+=i
print(summ)

Enter a number: 50
196


### Task 10
Write a Python code that will read 10 numbers from the user, and then print the first number, the sum of the first 2 numbers, the sum of the first 3 numbers, and so on up to the sum of 10 numbers.

==========================================================


For example,

The user enters 10, output 10, then \
The user enters 2, (10+2) = 12, output 12, then \
The user enters 4, (10+2+4) = 16, output 16, then \
The user enters 20, (10+2+4+20) = 36, output 36, and continues till the 10$^{th}$ input


In [1]:
#Todo
#taking sum as an empty integer
sum=0
for number in range(0,10):
    #taking 10 numbers from the user 
    number=int(input('enter a number: '))
    sum=sum+number #print input=1st,input=sum of 1st and 2nd, input=sum of 1st,2nd and 3rd and so on
    print(sum)
    

enter a number: 10
10
enter a number: 2
12
enter a number: 4
16
enter a number: 20
36
enter a number: 10
46
enter a number: 2
48
enter a number: 4
52
enter a number: 20
72
enter a number: 10
82
enter a number: 2
84


In [19]:
summ = 0 
for i in range(0,10):
    num = int(input('enter numbers: '))
    summ+=num
    print(summ)

enter numbers: 10
10
enter numbers: 2
12
enter numbers: 4
16
enter numbers: 20
36
enter numbers: 10
46
enter numbers: 2
48
enter numbers: 4
52
enter numbers: 20
72
enter numbers: 10
82
enter numbers: 2
84


# 

### Task 11
Write a Python program which takes a number and prints the digits from the unit place, then the tenth, then hundredth, etc. <font color='red'>(Right to Left)</font>

<font color='red'>[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]</font>

**Example:** If the user gives 32768, then print 8, 6, 7, 2, 3 

=========================================================================

**Hint (1):**  The input() function, converts the input data to String data type by default. Use this knowledge to solve this problem.


**Hint  (2):** First to get the digit from the right side, take the remainder of the number using modulus (%) operator and print the digit. For dropping the last digit, perform floor division by 10 on the number. Then start over.

32,768 % 10 = 8 

32,768 // 10 = 3, 276 

3, 276 % 10 = 6

3, 276 // 10 = 327 

327 % 10 = 7 

327 // 10 = 32 

32 % 10 = 2 

32 //10 = 3 

3 % 10 = 3 

3 // 10 = 0 

Done! When the number becomes 0 your loop should end.



In [1]:
#Todo
#taking a number from the user 
number=int(input('enter a number: '))
remainder=number%10 #32768%10=8 , to remove 8 from 32768
print(remainder,end='') #print out 8
while(number>0):
    number=number//10 #32768//10=3276 , to get 3276 from 32768 ans so on
    number1=number%10 #3276%10=6 , to remove 6 from 3276 and so on
    if(number1==0):
        break #stop loop 
    else:
        print(',',number1,end='')  #prints from right to left 

enter a number: 32768
8, 6, 7, 2, 3

In [14]:
num = int(input('Enter a number: '))
rem = num % 10
print(rem,end='')
ans = ''
while num>0 : 
    num=num//10
    rem2=num%10
    if rem2 == 0 :
        break 
    else:
        ans = ',' + str(rem2) 
    print(ans, end = '')


Enter a number: 86723
3,2,7,6,8

## num=int(input('enter:'))
list1=[]
ans=''
for i in str(num):
    list1.append(i)
for j in range(len(list1)-1,-1,-1):
    ans=list1[j]+','
    print(ans,end='')

### Task 12

Write a Python program that takes a number and tells how many digits are in that number. \
<font color='red'>[Consider the input number to be an INTEGER.]</font>

<font color='red'>You are not allowed to use len() function</font>

**Example:** If the user gives 9876, you should print 4. 



**Hint:** Keep floor dividing by ten and count how many times this could be divided. 

9876 floor divide by 10, is 987, count that got 1 digit (total 1) \
987 floor divide by 10, is 98, count that got 1 digit (total 2) \
98 floor divide by 10, is 9, count that got 1 digit (total 3) \
9 floor divide by 10, is 0, count that got 1 digit (total 4) 

Done! (When the number becomes 0 your loop should end.)



In [21]:
#Todo
#taking a number from the user
number=int(input('enter a number: '))
#initializing loop 
counter=0
#loop structure 
while (number>0):
    number=number//10
    counter=counter+1 #updating loop centre
    #inside loop body 
print('counter:',counter) #count the number of digits 
#outside loop body     

enter a number: 9876
counter: 4


In [4]:
number=int(input('enter a number: '))
counter=0
for i in str(number):
    counter+=1
print('counter:',counter)

enter a number: 9876
counter: 4


### Task 13
Write a Python program that takes a number from the user and prints its digits from left to right. <font color='red'> (left to right) </font>

<font color='red'>[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]</font>

=========================================================================

**Example:** if the user gives 32768, then print 3, 2, 7, 6, 8 

=========================================================================

**Hint(1):**  The input() function, converts the input data to String data type by default. Use this knowledge to solve this problem.

**Hint(2):**  

Step1: count how many digits\
Step2: calculate 10 to the power that (number of digits) minus 1.

Step3: Say, 32768 has 5 digits, so you calculate 10 to the power 4 which is 10,000. Then divide 32,768 by 10,000 and thus you get 3.\
Take remainder of 32,768 by 10,000 and thus you get 2,768 Then divide 10,000 by 10 to get 1,000\
Then divide 2,768 by 1,000 and thus you get 2.\
take remainder of 2,768 by 1,000 and thus you get 768 keep going on until there is no more digits left (zero!).


**In short:**\
Loop 1: First count digits, say 5 in this case for 32,768\
Loop 2: Then calculate 10 to the power 4 (5-1), that is 10,000.\
Loop 3: Then repeat following three steps\
32,768 // 10,000 = 3\
32,768 % 10,000 = 2,768 \
10,000//10 = 1,000

2,768 // 1,000 = 2\
2,768 % 1,000 = 768\
1,000//10 = 100

768 // 100 = 7\
768 % 100 = 68 \
100//10= 10

68 //10 = 6\
68 % 10 = 8 \
10//10= 1

8 //1 = 8\
8 % 1 = 0 \
1//10=0


In [2]:
#Todo
#taking number from the user
number=int(input('enter a number: '))
#storing number in number1
number1=number
#initializing counter
counter=0
#loop structure 1
while(number>0):
    number=number//10  
    counter=counter+1
#loop structure 2
while(number1>0):
    number=number1//10**(counter-1)
    print(number,end='')
    number1=number1%10**(counter-1)
    counter=counter-1
    if(number1!=0):
        print(end=',')
    
    

enter a number: 32768
3,2,7,6,8

In [64]:
num=int(input('enter:'))
count = 0 
for i in str(num):
    count += 1 
    if count == len(str(num)):
        print(i,end='')
    else:
        print(i,end=',')

enter:32768
3,2,7,6,8

### Task 14

Write a Python program that takes a number from the user and prints the divisors of that number and then print how many divisors were there.
<font color='red'>[The input number has to be an INTEGER]</font>

==========================================================

**Example01:** 

**Input:**\
6

**Output:**\
1, 2, 3, 6 \
Total 4 divisors.


==========================================================

**Example02:** 

**Input:**\
121

**Output:**\
1, 11, 121 \
Total 3 divisors.




In [11]:
#Todo
#taking a number from the user 
number=int(input('enter a number: '))
#taking counter and comma as empty integers 
counter=0
coma=0
for i in range(1,number+1): #range from 1 to number  
    if(number%i==0): #121%i==0 gives i which are the divisors of 121 when remainder is 0
        counter=counter+1 #incremating 
        if (coma!=0): #to remove comma from the end of loop.
            print(',',i,end='  ')
        else:
            print(i,end=' ')
        coma=coma+1 #incremating 
print() #to go to next row 
print('counter:',counter) #counting the number of divisors 

    

enter a number: 121
1 , 11  , 121  
counter: 3


In [83]:
num = int(input('enter a number: '))
count = 0 
count2 = 0 
for i in range(1,num+1):
    count2+=1
    if num % i == 0:
        count +=1
        print(i,end=',')
print('Total',count,'divisors')

enter a number: 121
1,11,121,Total 3 divisors


### Task 15
Write a Python program that takes a number and tells if it is a perfect number or not. <font color='red'>[The input number has to be an INTEGER]</font>

**Perfect Number:** An integer number is said to be a perfect number if its factors, including 1 but not the number itself, sum to the number.

==========================================================

**Example01:** 

**Input:**\
6

**Output:**\
6 is a perfect number

**Explanation:**\
6 have 4 divisors: 1, 2, 3, and 6. \
If we add all factors except itself, 6 = 1 + 2 + 3.

==========================================================


**Example02:** 

**Input:**\
28

**Output:**\
28 is a perfect number

**Explanation:**\
28 have 6 divisors: 1, 2, 4, 7, 14, and 28. \
If we add all factors except itself, 28 = 1 + 2 + 4 + 7 + 14.

==========================================================

**Example03:** 

**Input:**\
33

**Output:**\
33 is not a perfect number

**Explanation:**\
33 have 3 divisors: 1, 3, 11, and 33. \
If we add all factors except itself, 15 = 1 + 3 + 11.



In [22]:
#Todo
#taking a number from the user 
number=int(input('enter a number: '))
#taking sum as an empty integer 
sum=0
for i in range(1,number): #range from 1 to number-1
    divisor=number%i #6%i==0, to find the divisors of 6 when the remainder is 0
    if(divisor==0):
        sum=sum+i  #for the divisors for remainder=0, sum up 
if(sum==number):
    print(number,'is a perfect number') #sum=numper input, number is perfect
else:
    print(number,'is not a perfect number')
   
        

enter a number: 6
6 is a perfect number


In [88]:
num = int(input('enter a number: '))
summ = 0 
for i in range(1,num):
    if num % i == 0 :
        summ+=i
if summ == num:
    print(num,'is a perfect number')
else:
    print(num,'is not a perfect number')

enter a number: 6
6 is a perfect number


### Task 16

Write a Python program that asks the user for one number and tells if it is a prime number or not. <font color='red'>[The input number has to be an INTEGER]</font>

**Prime Number:** If a number has only two divisors, (1 and itself), then it is prime. If it is divisible by more numbers, then it is not a prime.

==========================================================

**Hint:** use the divisor count from task 16. 

==========================================================

**Example01:** 

**Input:**\
11

**Output:**\
11 is a prime number

**Explanation:**\
11 has only 2 divisors: 1, and 11.

==========================================================

**Example02:** 

**Input:**\
6

**Output:**\
6 is not a prime number

**Explanation:**\
6 have 4 divisors: 1, 2, 3 and 6.






In [42]:
#Todo
#taking a number from user 
number=int(input('enter a number: '))
#taking counter as an emoty integer 
counter=0
for i in range(1,number+1): #range from 1 to number 
    divisor=number%i
    if(divisor==0):  #6%i=0 to find all the divisors of 6 when remainder=0
        counter=counter+1 #incremating 
if(counter<=2):
    print(number,'is a prime number') #when numbers of  divisors (counter) are 2 or less
else:
    print(number,'is not a prime number') #when numbers of divisors are more than 2



enter a number: 6
6 is not a prime number


In [41]:
#extra 
number_list=input('enter a number:').split(',')
prime_list=[]
not_prime_list=[]
for number in range(1,len(number_list)):
    counter=0
    if(int(number_list[number])%number==0):
        counter+=1
    if(counter==0) and (number_list[number] not in prime_list):
        prime_list.append(int(number_list[number]))
    if(counter>0) and (number_list[number] not in not_prime_list):
        not_prime_list.append(int(number_list[number]))
my_dict={'Prime numbers':prime_list,'Not prime numbers':not_prime_list}
print('The list has prime numbers:',prime_list)
print('The list has not prime numbers:',not_prime_list)
print('Prime or not prime dictionary:',my_dict)

enter a number:5,7,8,9
The list has prime numbers: []
The list has not prime numbers: [7, 8, 9]
Prime or not prime dictionary: {'Prime numbers': [], 'Not prime numbers': [7, 8, 9]}


### Task 17

Write a Python program that asks the user for a quantity then takes that many numbers and prints the maximum, minimum and average of those numbers. 

**<font color='red'>[CANNOT USE MAX,MIN BUILT-IN FUNCTIONS][FOR THIS TASK YOU DO NOT NEED TO USE LISTS]</font>** 

==========================================================

**Example01:** If the user enters 5 as an input for quantity. Then enters 10, 4, -1, -100, and 1. \
Your program output should be: “Maximum 10”, “Minimum -100”, “Average is -17.2”

**Input:**\
5
10
4
-1
-100
1.

**Output:**\
Maximum 10\
Minimum -100\
Average is -17.2

**Explanation:**\
Average calculation: (10+4 + (-1) + (-100) + 1)/5 = -86/5=-17.2



In [2]:
#Todo
#taking sum and integer as empty integers 
sum=0
average=0
for i in range(0,5): #range from 0 to 4 for input 
    number=int(input('enter a number: '))  
    sum=sum+number
    average=sum/5
    if(i==0): #maximum=minimum=number
        maximum=number
        minimum=number
    elif(number>maximum): 
        maximum=number
    elif(number<minimum):
        minimum=number 
print('maximum' ,maximum)
print('minimum',minimum)
print('average',average)
        


enter a number: 10
enter a number: 4
enter a number: -1
enter a number: -100
enter a number: 1
maximum 10
minimum -100
average -17.2


In [100]:
num = int(input('enter a number: '))
maxim = 0 
minim = 9999999
summ=0
for i in range(0,num):
    nums = int(input('enter numbers: ')) 
    summ+=nums
    avg=summ/num
    if maxim < nums :
        maxim = nums 
    if minim > nums :
        minim = nums 

print('Maximum number: ',maxim)
print('Minimum number: ',minim)
print('Average: ',avg)

enter a number: 5
enter numbers: 10
enter numbers: 4
enter numbers: -1
enter numbers: -100
enter numbers: 1
Maximum number:  10
Minimum number:  -100
Average:  -17.2


In [6]:
number_list=[]
summ=0
count=0
for i in range(5):
    number=int(input('enter numbers:'))
    count+=1
    number_list.append(number)
    summ+=number
    average=summ/count
for num1 in range(0,len(number_list)):
    minimum=number_list[num1]
    minimum_index=num1
    for num2 in range(num1+1,len(number_list)):
        if(number_list[num2]<minimum):
            minimum=number_list[num2]
            minimum_index=num2
    temp=number_list[minimum_index]
    number_list[minimum_index]=number_list[num1]
    number_list[num1]=temp
print('Minimum number:',number_list[0])
print('Maximum number:',number_list[-1])
print('Average:',average)

enter numbers:10
enter numbers:4
enter numbers:-1
enter numbers:-100
enter numbers:1
Minimum number: -100
Maximum number: 10
Average: -17.2


## Task 18

Write a python program that prints a square of size N using + where N will be given as input.

=======================================

**Sample Input**
5

**Sample Output**<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>


=======================================

**Sample Input**
3

**Sample Output**<br>
+++<br>
+++<br>
+++<br>

In [1]:
#Todo
#taking a number from user 
N=int(input('enter a number: '))
for row in range(1,N+1): #range from 1 to N 
    for column in range(1,N+1): #range from 1 to N
        print('+',end='') #print + for column in a horizontal line 
    print(  ) #print + for row in vertical line 

enter a number: 5
+++++
+++++
+++++
+++++
+++++


In [101]:
N = int(input('Enter a number: '))
for i in range(1,N+1) : 
    for j in range(1,N+1):
        print('+',end='')
    print( )

Enter a number: 5
+++++
+++++
+++++
+++++
+++++


### Task 19

Write a python program that prints a rectangle of size M (height/line numbers) and N(length/column numbers) using incrementing numbers where M,N will be given as input.

===================================

**Sample Input**
4,6

**Sample Output**<br>
123456<br>
123456<br>
123456<br>
123456<br>


===================================


**Sample Input**
3,2

**Sample Output**<br>
12<br>
12<br>
12<br>


In [105]:
#Todo
#taking M and N from user 
M=int(input('enter a number: '))
N=int(input('enter a number: '))
for row in range(1,M+1): #range from 1 to M  #outside loop 
    for column in range(1,N+1): #range from 1 to N #inside loop 
        print(column,end='') 
    print()
    



enter a number: 2
enter a number: 4
1234
1234


In [104]:
N = int(input('enter a number for N: '))
M = int(input('enter a number for M: '))
for i in range(1,N+1):
    for j in range(1,M+1):
        print(j,end='')
    print( )

enter a number for N: 2
enter a number for M: 4
1234
1234


### Task 20

Write a python program that prints a right angled triangle of height N using incrementing numbers where N will be given as input.

====================================

**Sample Input**
4

**Sample Output**<br>
1<br>
12<br>
123<br>
1234<br>

====================================

**Sample Input**
5

**Sample Output**<br>
1<br>
12<br>
123<br>
1234<br>
12345<br>

In [9]:
#Todo
#taking a number from user 
number=int(input('enter a number: '))
for row in range(1,number+1): #range from 1 to number #outside loop
    for column in range(1,row+1): #range from 1 to row #inside loop 
        print(column,end='') 
    print('')


enter a number: 10
1
12
123
1234
12345
123456
1234567
12345678
123456789
12345678910


In [106]:
N = int(input('enter a number: '))
for i in range(1,N+1):
    for j in range(1,i+1):
        print(j,end='')
    print( )

enter a number: 3
1
12
123


## Optional Tasks (21 -26) [Ungraded]

### Task 21

Write a python program that prints all the fibonacci number from 0 to N where N will be given. 

**A Fibonacci number is a number which is the summation of its previous two fibonacci number.**

First two fibonacci number are 0 and 1. So the 3rd Fib will be 0+1=1, 4th Fib will be 1+1=2, 5th Fib will be 1+2=3 and so on.

==============================

**Sample Input**<br>
10

**Sample Output**<br>
0 1 1 2 3 5 8

==============================


**Sample Input**<br>
15

**Sample Output**<br>
0 1 1 2 3 5 8 13

In [2]:
#Todo
#taking a number from user  
number=int(input('enter a number: '))
#storing 1 and 0 in variables 
number1=0
number2=1
if(number==0): #if input is 0, print 0 only 
    print(number1)
else:  #if input is 1 => print 0 1 and if input is 3 => 0 1 1 and so on 
    print(number1,number2,end=' ')
#loop structure 
while(number2<=number): #input is greater than or equal to 1  
    sum=number1+number2
    if(sum<=number): #input is less than sum, so print sum 
        print(sum,end=' ')
    #updating variables 
    number1=number2 #sum=number2+number2 =>sum=1+1=2 (4th value in sequence)
    number2=sum #sum=sum+number2 => sum=2+1=3 (5th value in sequence)
    

enter a number: 10
0 1 1 2 3 5 8 

## Task 22

Write a python program that converts a Decimal Integer number to a Binary Number. 

**A decimal can be converted to a binary number by keeping track of the remainders after each division of that number by 2. **

================================

For example, to convert 10 to a binary number, we can follow the following approach

10/2 = 5 (Remainder 0) 
<br>5/2 = 2 (Remainder 1) 
<br>2/2 = 1 (Remainder 0) 
<br>1/2 = 0 (Remainder 1) 

Take the remainders from bottom to top, which is, 1010. Binary of 10 is 1010.

**Sample Input** <br>
10

**Sample Output**<br>
1010

================================

For example, to convert 13 to a binary number, we can follow the following approach

13/2 = 6 (Remainder 1) 
<br>6/2 = 3 (Remainder 0) 
<br>3/2 = 1 (Remainder 1) 
<br>1/2 = 0 (Remainder 1) 

Take the remainders from bottom to top, which is, 1101. Binary of 13 is 1101.

**Sample Input** <br>
13

**Sample Output**<br>
1101

In [ ]:
#Todo
number=int(input('enter a number: '))
remainder=''
while(number!=0):
    remainder+=str(number%2)
#     print(remainder)
    number=number//2
#     print(number)
print(remainder)
    

### Task 23

Write a python program that converts a Binary number to a Decimal Integer Number. 

**A binary number can be converted to its corresponding decimal number by multiplying each binary digit with a power of 2 where the power denotes the position of the binary digit**

========================================

For example, to convert 1010, we can do this -


$2^{3}$ x 1 + $2^{2}$ x 0 + $2^{1}$ x 1 + $2^{0}$ x 0 = 10


**Sample Input** <br>
1010

**Sample Output** <br>
10

========================================

For example, to convert 1101, we can do this -


$2^{3}$ x 1 + $2^{2}$ x 1 + $2^{1}$ x 0 + $2^{0}$ x 1 = 13


**Sample Input** <br>
1101

**Sample Output** <br>
13

In [13]:
#Todo
number=int(input('enter number: '))
counter=0
n=number
while(n>0):
    n=n//10
    counter=counter+1
#     print(n)
result=0
for i in range(counter):
#     print(i)
    a=number%10
#     print(a)
    number=number//10
#     print(number)
    result=result+(a*2**i)
    print(result)
# print(result)


    
    



enter number: 1101
1
1
5
13


In [1]:
number=input('enter a number: ')
newList=[]
lenght=len(number)
add=0
for i in range(len(number)):
    add=add+(2**(lenght-1))*int(number[i])
    lenght=lenght-1
print(add)

enter a number: 1101
13


### Task 24

Take five numbers from the user and find the minimum and the average of **only the even** numbers entered by the user. [If the user enters odd numbers ignore them] 

**<font color='red'>[CANNOT USE MAX,MIN BUILT-IN FUNCTIONS] [FOR THIS TASK YOU DO NOT NEED TO USE LISTS]</font>** 

Assume, the first input is always an even number.

============================================================

**Example01:** If the user enters 10, 4, -1, -100, and 1. \
Output: “Minimum -100”, “Average is 28.66667”

**Input:**\
10

4

-1

-100

1

**Output:**\
Minimum -100\
Average is 28.66667

**Explanation:**\
Average calculation: (10+4 + (-100))/3 = -86/3=-28.66667

============================================================

**Example02:** If the user enters 2, 10, 1, 21, and 3. \
Output: Minimum 2”, “Average is 6.0”

**Input:**\
2

10

1

21

3

**Output:**\
Minimum 2\
Average is 6.0

**Explanation:**\
Average calculation: (2+10)/2 = 12/2= 6





In [3]:
#Todo  
counter=0
sum=0
average=0
minimum = 99999999999999
for i in range(0,5):
    number=int(input('enter a number: '))
    if(number%2==0):
        counter=counter+1
        sum=sum+number
        average=sum/counter
        if(number<=minimum):
            minimum=number
print('minimum:',minimum)
print('average:',average)       

enter a number: 2
enter a number: 10
enter a number: 1
enter a number: 21
enter a number: 3
minimum: 2
average: 6.0


###### Task 25

Write a Python code for the following:
* Ask the user to enter a number, N (Total number of inputs to be taken). Assume that the user will never enter the first number as zero.
* Take N number of inputs
* Prints out the product of all the numbers read

For example, if the first input is 4, then the program has to read in four numbers from the user and print the product of these four numbers. 

=========================================================================

**Example01:**

**Input:**\
5 

10 

6 

4 

2 

1 

**Output:**\
480

**Explanation:**\
5 is the total number of inputs taken. Then, the calculation should be 10 X 6 X 4 X 2 X 1 = 480 







In [20]:
#Todo 
number=int(input('enter number of inputs: '))
multiply=1
for i in range(1,number+1):
    number1=int(input('enter a number: '))
    if(number1!=0):
        multiply=number1*multiply
        print(multiply)
        
   

enter number of inputs: 5
enter a number: 2
2
enter a number: 3
6
enter a number: 4
24
enter a number: 5
120
enter a number: 6
720


### Task 26
Ask the user for a range (a starting number and an ending number). 
**Count** how many numbers are prime numbers and how many numbers are perfect numbers between that range and **print** those numbers. 


=========================================================================

**Hint (1):** Declare two strings/lists to store the prime and perfect numbers. Inside the iteration store the values of the prime and perfect numbers in the pre- declared variables.

**Hint (2):**


ans = "Prime numbers: "\
prime_value = 11
ans = ans + str(prime_value) + " "\
print(ans)

ans = ans + str(13) + " "\
print(ans)

*Output:*

Prime numbers: 11 \
Prime numbers: 11 13 

=========================================================================

**Example:**between 2 and 6, there are 3 prime numbers (2, 3, 5) and 1 perfect number (6). 

**Input:**\
2

6

**Output:**\
Between 2 and 6,\
Found 3 prime numbers \
Found 1 perfect number\
Prime numbers: 2  3  5 \
Perfect numbers: 6 



In [32]:
#Todo
number1=int(input('enter a starting number: '))
number2=int(input('enter an ending number: '))
print('between',number1,'and',number2)
primeNumber=''
perfectNumber=''
counter=0
counter1=0
count=0
sum=1
for i in range(number1,number2+1): #range(2,6) #2,3,4,5,6 
#     print(i,end=' , ')
    for j in range(2,i): #2=>range(2,1)/3=>range(2,2)/4=>(2,3)
#         print(j,end=' , ')
        if(i%j==0): #i=2,no entry/i=3,no entry/i=4=>4%2=2,no entry 
            sum=sum+j #i=4=>is divisible by 2 only,so,sum=2
            count=count+1 
    if(sum==i): #i=2=>no entry/i=3,no entry/i=4,no entry[sum=2,i=4]/
        perfectNumber+= str(i) + ' '
        counter1=counter1+1
    if(count==0): #i=2,count remains 0,so,is a prime number/i=3,count remains 0,so,is a prime number/
        primeNumber+= str(i) + ' '
        counter=counter+1
    sum=1
    count=0

print('found',counter,'prime numbers')
print('found',counter1,'perfect numbers')
print('Prime numbers:',primeNumber)
print('Perfect numbers:',perfectNumber)

    



enter a starting number: 2
enter an ending number: 6
between 2 and 6
found 3 prime numbers
found 1 perfect numbers
Prime numbers: 2 3 5 
Perfect numbers: 6 
